### PR TITLE
[MINOR] EDA-1323 functional testing

### PR DESCRIPTION
**Pull request by srobledo-eb**
Creation of file 'functional_tests' inside 'tests' folder. It has several methods to:

- Test create_game method.
- Test different valid moves required by the server.
- Test invalid moves from the initial position of pawn.
- Test correct 'gameover' output once the last move is made.
- Test correct 'timeout' output after server sends this message.

### CODE REVIEW

File: tests/functional_tests.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        to_col,
+        expected_board,
+    ):
+        interface = Interface()
+        names = ['name1', 'name2']
+        start_response = await interface.create_game(names)
+        game_id = start_response.game_id
+        api_data = dict(
+            {
+                'action': 'move',
```
Comment (eb-cmoyano) [inline tests/functional_tests.py:241]: Please, can you use a constant?

File: tests/functional_tests.py [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        names = ['name1', 'name2']
+        start_response = await interface.create_game(names)
+        self.assertEqual(
+            start_response.turn_data['board'],
+            self.board_str['initial'],
+        )
+        game_id = start_response.game_id
+        api_data = dict(
+            {
+                'action': 'move',
```
Comment (eb-cmoyano) [inline tests/functional_tests.py:151]: Same here about the constant.

File: tests/functional_tests.py [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+        Test create_game method.
+    test_valid_moves():
+        Test different valid moves required by the server.
+    test_move_to_wrong_spot():
+        Test invalid moves from the initial position of pawn.
+    test_last_move():
+        Test correct 'gameover' output once the last move is made.
+    test_penalize_from_server():
+        Test correct 'timeout' output after server sends this message.
+    """
```
Comment (eb-cmoyano) [inline tests/functional_tests.py:31]: nit: Thanks for using `docstrings` but I think is not necessary here.

File: tests/functional_tests.py [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+        to_row,
+        to_col,
+    ):
+        interface = Interface()
+        names = ['name1', 'name2']
+        start_response = await interface.create_game(names)
+        game_id = start_response.game_id
+        api_data = dict(
+            {
+                'action': 'move',
```
Comment (eb-cmoyano) [inline tests/functional_tests.py:200]: same here about the constant.

File: tests/functional_tests.py [hunk 5]
```diff
@@ -0,0 +1,277 @@
+from unittest import IsolatedAsyncioTestCase
+from game.grpc_interfaz import Interface, MANAGER
+from edagames_grpc.game_start import GameStart
+from tests import VALID_STATE, INVALID_STATE, GAMEOVER_STATE, TIMEOUT, TOTAL_MOVES
+from parameterized import parameterized
```
Comment (eb-cmoyano) [inline tests/functional_tests.py:5]: 1st rule - (coding style) Sorting imports:
1. python built-in
2. external (third-party) dependencies
3. internal  dependencies

2nd Multi-import
```
from tests import VALID_STATE, INVALID_STATE, GAMEOVER_STATE, TIMEOUT, TOTAL_MOVES
```
->
```
from tests import (
    GAMEOVER_STATE,
    INVALID_STATE,
    TIMEOUT,
    TOTAL_MOVES,
    VALID_STATE,
)
```

TIP: Could you install https://pre-commit.com/ for the next time.

File: tests/functional_tests.py (hunk 1 above)
Comment (srobledo-eb) [reply tests/functional_tests.py]: I added a constant at the level of the 'tests' folder

File: tests/functional_tests.py (hunk 2 above)
Comment (srobledo-eb) [reply tests/functional_tests.py]: The same as the previous case

File: tests/functional_tests.py (hunk 3 above)
Comment (srobledo-eb) [reply tests/functional_tests.py]: I removed the documentation

File: tests/functional_tests.py (hunk 4 above)
Comment (srobledo-eb) [reply tests/functional_tests.py]: Solved with constant

File: tests/functional_tests.py (hunk 5 above)
Comment (srobledo-eb) [reply tests/functional_tests.py]: Applied standard order pattern

**Review by vgatica-eb — APPROVED** [review]

**Pull request merged**